madanalysis/configuration/weight_configuration.py

```diff
@@ -187,10 +189,14 @@ def nominal(self, scale_choice: int, central_pdfs: np.array) -> Weight:
                     continue
                 if not w.pdfset in central_pdfs:
                     continue
-                # !WARNING: this will fail if there are multiple pdfsets that are used
-                # ! this search will only return the first one
-                self._nominal = w
-        return self._nominal
+                weight_set.append(w)
+            if len(weight_set) > 1:
+                logging.getLogger("MA5").warning(
+                    "Variations on multi-pdf sets are not supported at the moment."
+                    f"{central_pdfs[weight_set[0].pdfset].name} will be used instead"
+                )
+
+        return weight_set[0]
```

Comment:
Can't we compute them independently and store all errors? Like for the multiple scale choice possibilities?

Reply:
we can, but lets go step by step please. First we need to figure which method to use for each pdfset then I can look into it.